src/main/kotlin/net/atos/zac/task/TaskService.kt

```diff
@@ -88,30 +88,20 @@ class TaskService @Inject constructor(
                 "with screen event resource ID: '$screenEventResourceId'."
         }
         val succesfullyAssignedTaskIds = mutableListOf<String>()
-        restTaakVerdelenGegevens.taken.forEach { restTaakVerdelenTaak ->
-            try {
-                flowableTaskService.readOpenTask(restTaakVerdelenTaak.taakId).let { task ->
-                    assignTaskAndOptionallyReleaseFromAssignee(task, restTaakVerdelenGegevens, loggedInUser)
-                    sendScreenEventsOnTaskChange(task, restTaakVerdelenTaak.zaakUuid)
-                }
-                succesfullyAssignedTaskIds.add(restTaakVerdelenTaak.taakId)
-            } catch (taskNotFoundException: TaskNotFoundException) {
-                LOG.log(
-                    Level.SEVERE,
-                    "No open task with ID '${restTaakVerdelenTaak.taakId}' found while assigning tasks. Skipping task.",
-                    taskNotFoundException
-                )
-            }
-        }
-        indexeerService.indexeerDirect(succesfullyAssignedTaskIds.stream(), ZoekObjectType.TAAK, true)
-        LOG.info { "Successfully assigned ${succesfullyAssignedTaskIds.size} tasks." }
+        try {
+            assignTasks(restTaakVerdelenGegevens, loggedInUser, succesfullyAssignedTaskIds)
+        } finally {
+            // always update the search index and send the screen event, also if exceptions were thrown
+            indexeerService.indexeerDirect(succesfullyAssignedTaskIds.stream(), ZoekObjectType.TAAK, true)
+            LOG.info { "Successfully assigned ${succesfullyAssignedTaskIds.size} tasks." }
```

Comment:
We don't really know if this was successful or failed because of one/several tasks, so for now perhaps just:
```suggestion
            LOG.info { "Assigned ${succesfullyAssignedTaskIds.size} tasks." }
```
We might as well return the number of tasks that were really assigned

Reply:
>We don't really know if this was successful or failed because of one/several tasks,

Yes we do? The `succesfullyAssignedTaskIds.add()` is done after the previous calls successfully complete. That is the whole idea.

Reply:
>We might as well return the number of tasks that were really assigned

You mean as part of the TAKEN_VERDELEN screen event sent to the browser? I agree but I suggest to leave that for a future story.

Reply:
That is why you now see on TEST e.g. `Successfully assigned 9 tasks.` when trying to assign 10 tasks. This is because one fails because it is a closed task.

Reply:
Great. Missed that part